internal/services/bot/bot_channel_web_chat_resource.go

```diff
@@ -166,15 +245,30 @@ func resourceBotChannelWebChatUpdate(d *pluginsdk.ResourceData, meta interface{}
 
 	channel := botservice.BotChannel{
 		Properties: botservice.WebChatChannel{
-			Properties: &botservice.WebChatChannelProperties{
-				Sites: expandSiteNames(d.Get("site_names").(*pluginsdk.Set).List()),
-			},
+			Properties:  &botservice.WebChatChannelProperties{},
 			ChannelName: botservice.ChannelNameBasicChannelChannelNameWebChatChannel,
 		},
 		Location: utils.String(azure.NormalizeLocation(d.Get("location").(string))),
 		Kind:     botservice.KindBot,
 	}
 
+	if !features.FourPointOhBeta() {
+		if d.HasChange("site_names") {
+			channel, _ := channel.Properties.AsWebChatChannel()
+			channel.Properties.Sites = expandSiteNames(d.Get("site_names").(*pluginsdk.Set).List())
+		}
+	}
+
+	if d.HasChange("site") {
+		channel, _ := channel.Properties.AsWebChatChannel()
+		channel.Properties.Sites = expandSites(d.Get("site").(*pluginsdk.Set).List())
+	}
+
+	if _, err := client.Update(ctx, id.ResourceGroup, id.BotServiceName, botservice.ChannelNameWebChatChannel, channel); err != nil {
+		return fmt.Errorf("updating %s: %+v", id, err)
+	}
+
+	// Unable to add a new site with block_user_upload_enabled, endpoint_parameters_enabled, no_storage_enabled in the same operation, so we need to make two calls
```

Comment:
Copy paste error?  I don't see a second call here

Reply:
The first call is at line 267. The second call is at line 272. Seems it's just a display problem.